src/services/herokuauth.service.js

```diff
@@ -123,6 +125,64 @@ export default class HerokuAuthService {
     return tokenCreated;
   }
 
+  /**
+   * First trades codes for a Heroku token.
+   * Then adds the Heroku token for a user. If the user already has a token, it will update the
+   * existing one.
+   * @param userId the user id to create a token for
+   * @param code the code sent by Heroku to create a token
+   * @param email the email tied to the heroku account
+   * @returns {Object} the token that was created or updated.
+   */
+  async getAndSetHerokuTokenForUser(userId, code, email) {
+    let res;
+    const secrets = config.get('oauth_secrets');
+    try {
+      res = await fetch(`${'https://cors-anywhere.herokuapp.com/' +
```

Comment:
This request should probably go in the `heroku.service.js` file

Reply:
Regarding all your comments about moving to .service, I'm not sure it makes sense because we're communicating with the services through fetches rather than through their libraries. To me it seems more clear to keep them in auth.